File: app/src/main/java/eu/caraus/dynamo/application/domain/udacity/TeaserVideo.java
package eu.caraus.dynamo.application.domain.udacity;


import com.google.gson.annotations.SerializedName;

import java.io.Serializable;


public class TeaserVideo implements Serializable {

	@SerializedName("vimeo_id")
	private String vimeoId;

	@SerializedName("youtube_url")
	private String youtubeUrl;

	@SerializedName("transcodings")
	private Transcodings transcodings;

	public void setVimeoId(String vimeoId){
		this.vimeoId = vimeoId;
	}

	public String getVimeoId(){
		return vimeoId;
	}

	public void setYoutubeUrl(String youtubeUrl){
		this.youtubeUrl = youtubeUrl;
	}

	public String getYoutubeUrl(){
		return youtubeUrl;
	}

	public void setTranscodings(Transcodings transcodings){
		this.transcodings = transcodings;
	}

	public Transcodings getTranscodings(){
		return transcodings;
	}

	@Override
 	public String toString(){
		return 
			"TeaserVideo{" + 
			"vimeo_id = '" + vimeoId + '\'' + 
			",youtube_url = '" + youtubeUrl + '\'' + 
			",transcodings = '" + transcodings + '\'' + 
			"}";
		}
}

File: app/src/main/java/eu/caraus/dynamo/application/data/domain/udacity/TeaserVideo.kt
package eu.caraus.dynamo.application.data.domain.udacity


import android.arch.persistence.room.*
import com.google.gson.annotations.SerializedName
import eu.caraus.dynamo.application.data.local.udacity.converters.TranscodingsTypeConverter
import io.reactivex.annotations.NonNull

import java.io.Serializable

@Entity(

        tableName = "teaser_video",

        indices = [ (Index("teaserId")) ],

        foreignKeys = [( ForeignKey(

                            entity = CoursesItem::class,

                            parentColumns = ["key"],

                            childColumns  = ["teaserId"],

                            onDelete      = ForeignKey.CASCADE )

                       )]

)
data class TeaserVideo  (

    @NonNull
    @PrimaryKey( autoGenerate = true )
    var teaserId : Int,

    @SerializedName("vimeo_id")
    var vimeoId: String? = null,

    @SerializedName("youtube_url")
    var youtubeUrl: String? = null,

    @SerializedName("transcodings")
    @TypeConverters( TranscodingsTypeConverter::class)
    var transcodings: Transcodings? = null

) : Serializable {
    constructor() : this(0,"","",null)
}